src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocStyleCheck.java

```diff
@@ -346,9 +349,16 @@ private void checkComment(final DetailAST ast, final TextBlock comment) {
      */
     private void checkFirstSentenceEnding(final DetailAST ast, TextBlock comment) {
         final String commentText = getCommentText(comment.getText());
+        final String[] sentences = commentText.split("\\.(?=\\s|$)");
+        String firstSentence = "";
+
+        if (sentences.length > 0) {
+            firstSentence = sentences[0];
+        }
 
         if (!commentText.isEmpty()
             && !endOfSentenceFormat.matcher(commentText).find()
+            && !inlineReturnTagPattern.matcher(firstSentence).find()
```

Comment:
```suggestion
            && !isInLineReturnTag
```

Reply:
Sorry but I didn't get it, what is the point of such updates? why use FP in this simple logic?

Reply:
code should be a English text, especially in long IF expressions. Non of like to read code for long to understand it, we try to come to model where we(any other contributr) read code quickly by junior engineer.

Reply:
yeah that's exactly why I think using FP will be more complex. it's ok for me to apply these changes, i'm just saying the original code is straight forward than the updated one. so please take another look at both and confirm if we really want to update the code.

Reply:
I like proposal of @nrmancuso .

Reply:
>Sorry but I didn't get it, what is the point of such updates? why use FP in this simple logic?

Good question :) My suggestion has nothing to do with using or not using functional programming. Here is why I have suggested such changes:

1. Maintainability. Our intention is immediately obvious to the next person to read this code. Imagine you come back to look at this code in a few months:

` final String[] sentences = commentText.split("\\.(?=\\s|$)");`

I see we want to split comment content by punctuation and whitespace into sentences

`String firstSentence = "";`

We are declaring a blank string for some reason...

```
        if (sentences.length > 0) {
            firstSentence = sentences[0];
        }
```

Ok, we want to get the first sentence and possibly reassign the `firstSentence` variable.

```
     if (!commentText.isEmpty()
            && !endOfSentenceFormat.matcher(commentText).find()
            && !inlineReturnTagPattern.matcher(firstSentence).find()
```

After re-reading these conditions a few times, I see that we want to use the inline return tag pattern matcher on the first sentence from the comment, and negate this. Ah - we want to make sure that the first sentence in this comment does not have an inline return tag!

Versus:

`final boolean hasInLineReturnTag...`

I don't need to read the rest of the stream operations. We want to know if the first sentence has an inline return tag. 

We visit the condition below:

```
        if (!commentText.isEmpty()
            && !endOfSentenceFormat.matcher(commentText).find()
            && !hasInLineReturnTag
```
We want to make sure this first sentence does NOT have an inline return tag.

2. Immutability. We should always prefer immutable data whenever possible.
3. Concision. We don't care about or need the result of any of the intermediate operations, just the final outcome.
4. From a high level, we want to know if some element of a collection matches our criteria. Java has an API for this; the Stream API. We should use it.

Reply:
@nrmancuso great, thanks for that!
now, do we need to extend the documentation examples in this PR? or should we open another issue for it?

Reply:
We should extend documentation in this PR.

Reply:
Done.
While updating the documentation I found this wrong [Example](https://checkstyle.org/checks/javadoc/javadocstyle.html#:~:text=private%20void%20methodWithValidCommentStyle()%20%7B%7D%0A%0A%20%20%20%20/**-,*%20Some%20description%20here%20//%20violation%2C%20the%20sentence%20must%20end%20with%20a%20proper%20punctuation,-*/%0A%20%20%20%20private%20void%20methodWithInvalidCommentStyle), this doesn't produce a violation and I don't know if it was intended or not, anyway I updated it in the documentation for now.

Reply:
Please restore what was there, and let's open a new issue for this. We can mention that documentation is misleading and that examples need to be updated in the new issue.

Reply:
Done at https://github.com/checkstyle/checkstyle/issues/14510